modules/home-environment.nix

```diff
@@ -578,6 +578,7 @@ in
           pkgs.gnugrep
           pkgs.gnused
           pkgs.ncurses          # For `tput`.
+          pkgs.nix
```

Comment:
This package is incompatible with nix flakes.

I think the package should use whatever `nix` is available in the PATH to work in all scenarios.

Reply:
> This package is incompatible with nix flakes.

Do you mean "home-manager" is incompatible with flakes? That would seem weird to me since home-manager offers a flake.nix...

Reply:
No, currently `pkgs.nix` is the version 2.3 of Nix and it does not support flakes at all because the `experimental-features` option and flag is unavailable.

Reply:
Sorry, missed this. Yeah, `pkgs.nix` is not sufficient, need to match the system one. In the nixos module we are able to pick the right one from the system configuration but for standalone HM installs we can't really tell, we just have to rely on the user having a good Nix install.